test/metabase/db/custom_migrations_test.clj

```diff
@@ -1981,55 +1981,57 @@
                                   :uploads_table_prefix nil}}
                   (m/index-by :id (t2/select :metabase_database)))))))))
 
+(defn- sample-content-created? []
+  (boolean (not-empty (t2/query "SELECT * FROM report_dashboard where name = 'E-commerce Insights'"))))
+
 (deftest create-sample-content-test
   (testing "The sample content is created iff *create-sample-content*=true"
     (doseq [create? [true false]]
       (testing (str "*create-sample-content* = " create?)
-        (impl/test-migrations "v50.2024-05-27T15:55:22" [migrate!]
-          (let [sample-content-created? #(boolean (not-empty (t2/query "SELECT * FROM report_dashboard where name = 'E-commerce insights'")))]
-            (binding [custom-migrations/*create-sample-content* create?]
-              (is (false? (sample-content-created?)))
-              (migrate!)
-              (is ((if create? true? false?) (sample-content-created?))))
-
-            (when (true? create?)
-              (testing "The Examples collection has permissions set to grant read-write access to all users"
-                (let [id (t2/select-one-pk :model/Collection :is_sample true)]
-                  (is (partial=
-                       {:collection_id id
-                        :perm_type     :perms/collection-access
-                        :perm_value    :read-and-write}
-                       (t2/select-one :model/Permissions :collection_id id)))))))))))
+        (impl/test-migrations "v52.2024-12-03T15:55:22" [migrate!]
+          (binding [custom-migrations/*create-sample-content* create?]
+            (is (false? (sample-content-created?)))
+            (migrate!)
+            (if create?
+              (is (true? (sample-content-created?)))
+              (is (false? (sample-content-created?)))))
```

Comment:
Have to have at least one subjective nit

```suggestion
            (is (= create? sample-content-created?))
```

Reply:
Haha, thanks. My goal was to make the test print out what failed, instead of a report about:

`((if create? true? false?) (sample-content-created?)) failed` 

I'll take your suggestion since I think it does that too.